src/librustdoc/clean/types.rs

```diff
@@ -404,6 +404,16 @@ pub(crate) fn rustc_span(def_id: DefId, tcx: TyCtxt<'_>) -> Span {
 }
 
 impl Item {
+    /// If `item.name` is `None` and `self` is an `ImportItem`, it'll look for it.
+    ///
+    /// It's especially usefully to get the name of `pub use x;` or `pub use x as y;`.
+    pub(crate) fn full_name(&self) -> Option<Symbol> {
+        self.name.or_else(|| {
+            if let ImportItem(ref i) = *self.kind &&
+            let ImportKind::Simple(s) = i.kind { Some(s) } else { None }
```

Comment:
Does `s` refer to the `y` or the `x` in `pub use x as y;`?

Reply:
Yes. `Simple` is created with `new_simple` which returns an `Import`. The `Import` struct contains the `source` (so `x` in this case) and the kind (our `Simple("y")`).